### PR TITLE
Add telegraf operator helm chart to the requirements

### PR DESCRIPTION
**Pull request by sumo-drosiek**
###### Description

Add telegraf operator helm chart to the requirements

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: vagrant/values.yaml [hunk 1]
```diff
@@ -23,3 +23,6 @@ sumologic:
   accessId: dummy
   accessKey: dummy
   endpoint: http://receiver-mock.receiver-mock:3000/terraform/api/
+telegraf:
+  enabled: true
+  replicaCount: 1
```
Comment (frankreno) [inline vagrant/values.yaml:28]: If we have more than 1 replica do things actually work? Is there a need, perhaps cuz of load, to have more than 1?

**Review by frankreno — APPROVED** [review]
Some comments, otherwise LGTM

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -1412,3 +1412,49 @@ otelcol:
...
+## ref: https://github.com/influxdata/helm-charts/blob/master/charts/telegraf-operator/values.yaml
+telegraf:
+  enabled: false
+  replicaCount: 1
+  classes:
+    secretName: "telegraf-operator-classes"
+    default: "sumologic-prometheus"
+    data:
+      sumologic-prometheus: |
+        [[outputs.prometheus_client]]
```
[the remark this answers is not in the document]
Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:1427]: Updated

File: vagrant/values.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply vagrant/values.yaml:28]: Yes they will work. Didn't tested the telegraf-operator in terms of performance, but will change the replicaCount to 3 (as it's default value)

Comment (frankreno) [reply vagrant/values.yaml:28]: I am fine with 1. In my experience, there is almost always 1 operator.  I just wanted to understand the risks if customers scale.  eg will it work.

**Review by frankreno — APPROVED** [review]
LGTM

**Review by pmalek-sumo — APPROVED** [review]

File: deploy/helm/sumologic/values.yaml [hunk 3]
```diff
@@ -1537,3 +1537,20 @@ otelcol:
           receivers: [jaeger, opencensus, otlp, zipkin]
           processors: [memory_limiter, k8s_tagger, source, resource, batch, queued_retry]
           exporters: [zipkin]
+
+## Configure telegraf-operator
+## This is an experimental configuration and shouldn't be used in production environment
+## ref: https://github.com/influxdata/helm-charts/blob/master/charts/telegraf-operator/values.yaml
+telegraf-operator:
+  enabled: false
+  replicaCount: 1
```
Comment (perk-sumo) [inline deploy/helm/sumologic/values.yaml:1546]: @sumo-drosiek what is the replicatCount about?

**Review by perk-sumo — COMMENTED** [review]
One comment only, otherwise LGTM :)

File: deploy/helm/sumologic/values.yaml (hunk 3 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:1546]: about replicas 🤷 . It's 3 by default. AFAIK telegraf-operator performs leader election

**Pull request merged**